services/headless-lms/utils/src/url_to_oembed_endpoint.rs

```diff
@@ -42,6 +66,15 @@ pub fn url_to_oembed_endpoint(url: String) -> Result<Url, UtilError> {
                 &format!("url={}&maxwidth=780&maxheight=440", url),
             );
         }
+        if host.ends_with("menti.com") || host.ends_with("mentimeter.com") {
+            return oembed_url_builder(
+                &format!(
+                    "{}/api/v0/cms/gutenberg/oembed/mentimeter",
+                    base_url.unwrap()
+                ),
+                &format!("url={}", encode(url.as_str())),
```

Comment:
The set_query is used in the `oembed_url_builder`, but unfortunately it doesn't encode.

The problem is that we need to encode the query_parameter named `url`as this query parameter has its own query parameters (i.e. http://something.com?height=500&title=something) and when we build the whole url (http://project-331.local/api/v0/.../mentimeter?url=(endoded)) and later parse it. This way it will later when we parse the query param recognize its query params correctly.

Reply:
So, I understood from your description that the problem was that the `set_query` function does not understand that the  `&` sign in the something.com URL does not start the next query parameter, but just belongs to the previous one. Thankfully, the URL library has a function that handles this case correctly, too: https://docs.rs/url/2.2.2/url/struct.Url.html#method.query_pairs_mut.

I now see that the reason why did this this way was because the `oembed_url_builder` signture is like this: 
```rust
fn oembed_url_builder(url: &str, query_params: &str) -> Result<Url, UtilError>
```
which kindof encourages to do stuff this way. One way to make this cleaner could be to change functions signature to somethign like this:

```rust
fn oembed_url_builder(url: &str, query_params: &[(&str, &str)]) -> Result<Url, UtilError>
```
Then we could interate through the slice and use `query_pairs_mut`.

Reply:
If we really wanted to do the escaping ourselves, I'd use the same encoding library as the URL library uses: https://crates.io/crates/percent-encoding

The point is to minimize the number of dependencies and we already transitively depend on this one.

Reply:
Swapped to the transitive dependency.